Almost done, but I have 2 questions

tzatushevskaya opened the pull request:
Approve and then I'll ask ;)

Commit:
- created several new views
 - added each task on the view with its name
 - included them in router file
 - mentioned them on the view containing links to each sub-view
 - updated App.vue
 - updated README.md
 - moved asserts

eclimov commented on src/views/Sjofn/Homework1.vue:
```diff
@@ -0,0 +1,171 @@
+  methods: {
+    result () {
+      let a = document.getElementById('a').value
+      let b = document.getElementById('b').value
```
Please, use data-attributes and `v-model` approach here.

eclimov commented on src/views/Sjofn/Homework4.vue:
```diff
@@ -0,0 +1,166 @@
+  },
+  methods: {
+    negate (bool) {
+      // return !bool
```
Please, remove redundant commented code.

eclimov commented on src/views/Sjofn/Homework4.vue:
```diff
@@ -0,0 +1,166 @@
+        'eyeColor': this.eyeColor,
+        'hobbies': this.hobbies
+      }
+      console.log(user)
```
You can avoid creating extra variable here and print the value directly:
`console.log({
   'name': this.name,
   'age': this.age,
    'gender': this.gender,
    'eyeColor': this.eyeColor,
    'hobbies': this.hobbies
})`

eclimov commented on src/views/Sjofn/Practice41.vue:
```diff
@@ -0,0 +1,56 @@
+      one: '',
+      two: '',
+      three: '',
+      four: ''
```
You don't need any of these data attributes, because all of methods (or 'functions', if you wish) don't work with any values from component's scope. Each method either receives some value or creates it inside.
Please, remove the data-attributes and check that everything still works as expected.

P.S.:
please, also check this article: https://learn.javascript.ru/function-basics

eclimov commented on src/views/Sjofn/Practice42.vue:
```diff
@@ -0,0 +1,94 @@
+      l: '',
+      p: '',
+      min: '',
+      max: ''
```
**Please, don't create a data-attribute for every value you use.** 
`data` is a section when you define values (variables, mostly) for **the whole component's scope**. 
Many of these are redundant, never used and should be removed.

For example, let's check 'z' value. 
It is an argument of `getPowFive` method. Everything you pass into the mthod transforms into local variable 'z'. The method doesn't require any dependencies from outer scope. Why you added 'z' in `data` then?
Please, remove it, check other and remove all redundant ones.

eclimov replied:
Contact me, if it is not clear.

eclimov commented on src/views/Sjofn/Practice42.vue:
```diff
@@ -0,0 +1,94 @@
+    this.genres.push('Humour')
+    this.disciplines.push('Drawing')
+    console.log(this.genres.join(', ') + ', ' + this.disciplines.join(', '))
+    this.fictionCharacter['discipline'] = 'physical education' // мне кажется, здесь что-то не так. Не нравится, как отображается в консоли(
```
Regarding your comment. 
Yes, it is not right, because your `fictionCharacter` value is an array of objects, while it should be an object.

eclimov replied:
Please, read again about objects and arrays. The difference is **very important to understand** in context of programming.
https://www.w3schools.com/js/js_objects.asp
https://www.w3schools.com/js/js_arrays.asp

eclimov replied:
Remove your comment in code, when done.

eclimov commented on src/views/Sjofn/Practice42.vue:
```diff
@@ -0,0 +1,94 @@
+    console.log(this.randomizePrice(100, 2000))
+  },
+
+  methods: {
```
As fa as I remember, you didn't need any methods in this practical task. 
Please, check `README.md`, - it is our 'source of truth'.
Remove redundant code.

eclimov commented on src/views/Sjofn/Practice43.vue:
```diff
@@ -0,0 +1,74 @@
+      name: '',
+      min: '',
+      max: '',
+      a: ''
```
Do not create `data` values for local variables.

P.S.: 'local variable' means a value that is not visible outside of block of code, like an argument of a function or something declared **within** function. 
Please, read and learn the following:
- Javascript variables
- Javascript functions and parameters
- Javascript "pure function"
https://medium.com/@jamesjefferyuk/javascript-what-are-pure-functions-4d4d5392d49c

eclimov replied:
Remove redundant `data` values.

Commit:
- removed redundant data
 - removed commented code
 - updated "export default { name: 'Practice3' > Practice43" etc.

eclimov commented on src/views/Sjofn/Homework4.vue:
```diff
@@ -0,0 +1,158 @@
+      console.log(`${adjective} adventures of ${characterName}`)
+    },
+
+    buildUserModel () {
```
Function's [signature](https://en.wikipedia.org/wiki/Type_signature) should correspond version from the html file. 
Please, recheck and fix.

eclimov commented on src/views/Sjofn/Practice42.vue:
```diff
@@ -0,0 +1,84 @@
+        ceilingsMinHeight: 5
+      },
+      fictionCharacter:
+        {
```
Please, use the same code style as you used for `livingRoomDesign` object (opening bracket on the same line and, closing one - on the same level).

eclimov commented on src/views/Sjofn/Practice41.vue:
```diff
@@ -0,0 +1,47 @@
+    currentDate.getFullYear() + ' ' +
+    currentDate.getHours() + ':' +
+    currentDate.getMinutes() + ':' +
+    currentDate.getSeconds()
```
Fix code style for contents of `getDatetime` method.

eclimov commented on src/views/Sjofn/Practice41.vue:
```diff
@@ -0,0 +1,47 @@
+  methods: {
+    getDatetime () {
+      let currentDate = new Date()
+      let dateTime = currentDate.getDate() + '/' +
```
Use template literals instead of string concatenation.

eclimov commented on src/views/Sjofn/Practice42.vue:
```diff
@@ -0,0 +1,84 @@
+  created () {
+    this.genres.push('Humour')
+    this.disciplines.push('Drawing')
+    console.log(this.genres.join(', ') + ', ' + this.disciplines.join(', '))
```
Use template literals instead of string concatenation.

Commit:
- inserted v-model inputs in Homework1

eclimov commented on src/views/Sjofn/Homework1.vue:
```diff
@@ -125,11 +120,17 @@
-      alert('The result is ' + Math.sqrt(Math.pow(parseInt(a), 2) + Math.pow(parseInt(b), 2)))
+    result (a, b) {
+      // let a = document.getElementById('a').value
+      // let b = document.getElementById('b').value
```
Remove commented code

Commit:
- corrected issues
 - removed methods where not required
 - updated code style
 - updated functions signature

eclimov commented on src/views/Sjofn/Homework4.vue:
```diff
@@ -0,0 +1,164 @@
+
+    // buildUserModel (userName, age, gender, eyeColor, hobbies) {
+    //   console.log(`${userName}, ${age}, ${gender}, ${eyeColor}, ${hobbies}`)
+    // },
```
Remove commented code.

Commit:
- excluded maps from public
 - removed commented code

Commit:
- updated gitignore

SandaPanteleimonov reviewed (approved):
Approved

Commit:
Merge branch 'master' into sjofn-hw5

# Conflicts:
#	public/hw2-united.html
#	src/App.vue
#	src/assets/palette.png
#	src/router/index.js
#	src/views/sanda/Homework3.vue

Commit:
-  |

eclimov reviewed (approved):
Good job!
